docs/minikube-tutorial.md

```diff
@@ -0,0 +1,201 @@
+
+```
+# change 127.0.0.1:1086 to your http/https proxy server IP:PORT
+minikube start --docker-env https_proxy=http://127.0.0.1:1086 \
+  --docker-env http_proxy=http://127.0.0.1:1086
+```
+
+See [minikube setup](https://kubernetes.io/docs/setup/minikube/) for more options to
+configure your virtual machine and Kubernetes cluster.
+
+### Install kubectl to access the cluster
+
+The Kubernetes command-line tool,
+[kubectl](https://kubernetes.io/docs/user-guide/kubectl/), allows you to run
+commands against Kubernetes clusters.
+
+Install kubectl according to the instructions in [Install and Set Up kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/).
+
+After kubectl is installed, test your minikube Kubernetes cluster:
+
+```
+kubectl cluster-info
+```
+
+## Install TiDB operator and run a TiDB cluster with it
+
+### Install helm
+
+Helm is the package manager for Kubernetes and is what allows us to install all of the distributed components of TiDB in a single step. Helm requires both a server-side and a client-side component to be installed.
+
+```
+curl https://raw.githubusercontent.com/helm/helm/master/scripts/get | bash
+```
+
+Install helm tiller:
+
+```
+helm init
+```
+
+If you have limited access to gcr.io, you can try a mirror, e.g.
+
+```
+helm init --upgrade --tiller-image registry.cn-hangzhou.aliyuncs.com/google_containers/tiller:$(helm version --client --short | grep -P -o 'v\d+\.\d+\.\d')
+```
+
+Once it is installed, running `helm version` should show you both the client
+and server version, e.g.
+
+```
+$ helm version
+Client: &version.Version{SemVer:"v2.13.1",
+GitCommit:"618447cbf203d147601b4b9bd7f8c37a5d39fbb4", GitTreeState:"clean"}
+Server: &version.Version{SemVer:"v2.13.1",
+GitCommit:"618447cbf203d147601b4b9bd7f8c37a5d39fbb4", GitTreeState:"clean"}
+```
+
+If it shows only the client version, `helm` cannot yet connect to the server. Use
+`kubectl` to see if any tiller pods are running.
+
+```
+kubectl -n kube-system get pods -l app=helm
+```
+
+### Install TiDB operator in the Kubernetes cluster
+
+Clone tidb-operator repository:
+
+```
+git clone https://github.com/pingcap/tidb-operator
+cd tidb-operator
+kubectl apply -f ./manifests/crd.yaml
+helm install charts/tidb-operator --name tidb-operator --namespace tidb-admin
+```
+
+Now, we can watch the operator come up with:
+
+```
+watch kubectl get pods --namespace tidb-admin -o wide
+```
+
+If you have limited access to gcr.io (pods failed with ErrImagePull), you can
+try a mirror of kube-scheduler image. You can upgrade tidb-operator like this:
+
+```
+helm upgrade tidb-operator charts/tidb-operator --namespace tidb-admin --set \
+  scheduler.kubeSchedulerImageName=registry.cn-hangzhou.aliyuncs.com/google_containers/kube-scheduler
+```
+
+When you see both tidb-scheduler and tidb-controller-manager are running, you
+can process to launch a TiDB cluster!
+
+### Launch a TiDB cluster
+
+```
+helm install charts/tidb-cluster --name tidb-cluster --set \
```

Comment:
It will be easier for the user to make helm changes if we have a minikube-values.yaml file. So just one option `--values minikube-values.yaml`

Reply:
Setting values from command line makes it easier for users to copy/paste. Besides, if use a separate `values.yaml`, we have to keep it in sync with the default `values.yaml`.

Reply:
`minikube-values.yaml` would have just the values to override, it wouldn't contain all the values.